mage_ai/data_preparation/models/pipeline.py

```diff
@@ -198,6 +207,54 @@ def create(self, name, pipeline_type=PipelineType.PYTHON, repo_path=None):
         )
         return pipeline
 
+    @classmethod
+    def import_from_zip(self, zip_content, overwrite=False):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            zip_data = BytesIO(zip_content)
+            with zipfile.ZipFile(zip_data, 'r') as zipf:
+                zip_size = sum(e.file_size for e in zipf.infolist())  # calc zip size in bytes
+                if zip_size / 1000 > PIPELINE_MAX_SIZE:  # prevention against zip-bombs
+                    raise PipelineZipTooLargeError(
+                        f'Pipeline zip excedes size limit {PIPELINE_MAX_SIZE/1000}Kb')
+
+                zip_contents = zipf.namelist()
+                prefix = os.path.commonpath(zip_contents)  # Check if zip contents are part of a root folder
```

Comment:
The "line too long" linting error causing the CI/CD tests to fail may be due to this line.